src/logic/safe/transactions/gas.ts

```diff
@@ -216,8 +217,13 @@ export const estimateGasForTransactionCreation = async (
       to: safeAddress,
       from: safeAddress,
       data: estimateData,
+      gas: safeTxGas ? safeTxGas : undefined,
     })
 
+    if (safeTxGas) {
```

Comment:
are we assuming that safeTxGas will always be enough? That's because it comes from the backend, isn't it?

Reply:
That value comes from the user, if the operation:
```
    const gasEstimationResponse = await getGasEstimationTxResponse({
      to: safeAddress,
      from: safeAddress,
      data: estimateData,
      gas: safeTxGas ? safeTxGas : undefined,
    })
```
Is successful with that amount of gas that means that the transaction will be ok with that amount of gas and we can return it. In case we don't have a manual value of gas we are trying to estimate how much we would need for that and we continue with the next line